### PR TITLE
Enable building of RPM and DEB packages with CPack

### PR DESCRIPTION
**Pull request by yuumasato**
Build RPM and DEB packages using CPack.

Packages are not identical to the one generated using the included spec file.
I couldn't remove "Linux" from the name of the generated rpm and debs, for example.

The RPMs installs nicely, but I didn't test the DEB.

### CODE REVIEW

**Comment by redhatrises** [comment]
@yuumasato why re-work to use the existing spec file implementing it with something like:
```
configure_file("${CMAKE_CURRENT_SOURCE_DIR}/scap-security-guide.spec.in" "${CMAKE_CURRENT_BINARY_DIR}/scap-security-guide.spec" @ONLY IMMEDIATE)
set(CPACK_RPM_USER_BINARY_SPECFILE "${CMAKE_CURRENT_BINARY_DIR}/scap-security-guide.spec")
```

**Comment by mpreisler** [comment]
@redhatrises IMO this is better than using the `spec.in` file.

**Comment by yuumasato** [comment]
At first it looked simpler to use the spec file, but when I got hands on it became ugly.

CPack setups the buildroot with `SOURCES` directory empty, it doesn't know where to get the tarball, and looking at the source there is no variable to specify where the tarball is.
So I had to change the `%prep` to manualy get the tarball and unpack it at the right place.
Also added `package_source` as a dependency of `package`, because CPack doesn't get or prepare the source for itself.

I managed to make improvements on CPack configuration and the RPMs look exactly like the ones generated through `Makefile`.

@shawndwells Do you think that building RPMs configured solely by CPack would satisfy your use case?  Would you still need to maintain the spec file upstream?

**Comment by pthierry38** [comment]
Hi all ! 

I'm not sure it is quite easy to pack properly deb packages (as discussed in #1640). To support the deb format, I've uploaded scap-security-guide as official Debian package for sid with all the Deb-specific constraints (https://tracker.debian.org/pkg/scap-security-guide).

I'll manage this package and update it to 0.1.32 during next weeks (have to resolve #1720 first). It will appear in Ubuntu and in next Debian testing with time.

**Comment by shawndwells** [comment]
@yuumasato end-game is to 'yum localinstall' an RPM from upstream. Make vs CMake isn't particularly important to me. Whatever you think is the cleanest approach!

File: CMakeLists.txt [hunk 1]
```diff
@@ -220,4 +221,53 @@ set(CPACK_SOURCE_IGNORE_FILES
...
+set(CPACK_RPM_DOC_PACKAGE_REQUIRES "scap-security-guide = ${SSG_VERSION}")
+set(CPACK_RPM_DOC_PACKAGE_SUMMARY "HTML formatted documents containing security guides generated from XCCDF benchmarks.
+")
+set(CPACK_RPM_DOC_PACKAGE_DESCRIPTION "The %{name} package contains HTML formatted documents containing
+hardening guidances that have been generated from XCCDF benchmarks
+present in scap-security-guide package.
+")
+
+
+set(CPACK_GENERATOR "RPM;DEB")
```
Comment (redhatrises) [inline CMakeLists.txt]: @yuumasato let's remove DEB for now and let @pthierry38 work out the Debian building.

Comment (mpreisler) [inline CMakeLists.txt]: No, why? Let's treat DEB building and RPM building are the same. We build both downstream and upstream. Please let's keep deb biulding here, there is no reason to remove it.

**Comment by mpreisler** [comment]
> I'm not sure it is quite easy to pack properly deb packages (as discussed in #1640). To support the deb format, I've uploaded scap-security-guide as official Debian package for sid with all the Deb-specific constraints (https://tracker.debian.org/pkg/scap-security-guide).

That's fine. We do the same thing with RPMs. The upstream RPM and DEB are meant mostly for testing and not for real deployment. We are aware that they are limited and even broken for a lot of use-cases. I have pushed in the past to remove them from upstream altogether but they do have legitimate uses so we kept them.

File: CMakeLists.txt (hunk 1 above)
Comment (redhatrises) [inline CMakeLists.txt]: I was wanting @pthierry38 would get it working as close to what he would like, but okay.

**Comment by mpreisler** [comment]
ACK

**Comment by pthierry38** [comment]
Sorry to respond that late (holidays time)

> That's fine. We do the same thing with RPMs. The upstream RPM and DEB are meant mostly for testing and not for real deployment. We are aware that they are limited and even broken for a lot of use-cases. I have pushed in the past to remove them from upstream altogether but they do have legitimate uses so we kept them.

ok I understand that use-case.

**Pull request merged**